Face-Mask-Detection/face_mask_detection_using_cnn.py

```diff
@@ -0,0 +1,136 @@
+import numpy as np
+import pandas as pd
+import cv2
+import matplotlib.pyplot as plt
+from tensorflow.keras.models import Model, Sequential
+from tensorflow.keras import layers
+import tensorflow as tf
+from keras.preprocessing.image import img_to_array
+from tensorflow.keras import backend
+from tensorflow.keras.preprocessing.image import ImageDataGenerator
+import os
+from sklearn.metrics import classification_report
+import sklearn.metrics as metrics
+import itertools
+for dirname, _, filenames in os.walk('/kaggle/input'):
+    for filename in filenames:
+        print(os.path.join(dirname, filename))
+def data_set(dir_data):
+    data=[]
+    target=[]
+    data_map = {
+    'with_mask':1,
+    'without_mask':0
+    }
+    skipped=0
+    root=dir_data+'_annotations.csv'
+    df1 = pd.read_csv(root)
+    df1.dataframeName = '_annotations.csv'
+    nRow, nCol = df1.shape
+    for i in range(len(df1)):
+        without_mask='without_mask'
+        k=dir_data+df1['filename'][i]
+        image=cv2.imread(k)
+        xmin=int(df1['xmin'][i])
+        ymin=int(df1['ymin'][i])
+        xmax=int(df1['xmax'][i])
+        ymax=int(df1['ymax'][i])
+        image=image[ymin:ymax,  xmin:xmax]
+        try:
+                # resizing to (70 x 70)
+                image = cv2.resize(image,(70,70))
+        except Exception as E:
+                skipped += 1
+                print(E)
+                continue
+        if(df1['class'][i]=='mask'):
+            without_mask='with_mask'
+        image=img_to_array(image)
+        data.append(image)
+        target.append(data_map[without_mask])
+    data = np.array(data, dtype="float") / 255.0
+    target = tf.keras.utils.to_categorical(np.array(target), num_classes=2)
+    return data, target
+training_target=data_set('./Face-Mask-Detection/kaggle/input/face-mask-detection/train/')
```

Comment:
What do you want to load here? There is no extension to files. Same for next testing_target and valid_target. Also, where are the testing_data, training_data, and valid_data variables? They are not initialized anywhere but passed into functions.

Reply:
@kaustubhgupta I think he is specifying the file [here](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/a6c083ed747952a453825cbdfa022522eb1bcc6e/Face-Mask-Detection/face_mask_detection_using_cnn.py#L23-L25) , I could be wrong as I haven't run this code yet

Reply:
Actually the pylint test gave these errors and when I saw the script, I found these variables are missing but passed in functions

Reply:
@kaustubhgupta Yup , the linter is giving error because of undefined variables. I am referring to the training_target variable , he is taking directory as argument to data_set function and then appending file name [here](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/a6c083ed747952a453825cbdfa022522eb1bcc6e/Face-Mask-Detection/face_mask_detection_using_cnn.py#L23-L25).

Reply:
> What do you want to load here? There is no extension to files. Same for next testing_target and valid_target. Also, where are the testing_data, training_data, and valid_data variables? They are not initialized anywhere but passed into functions.

extension is not required, because they are folders and they will directly have access to the files within them.